app/src/main/java/com/ozonehis/eip/ShutdownServer.java

```diff
@@ -0,0 +1,10 @@
+package com.ozonehis.eip;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class ShutdownServer {
+    public void shutdown() {
+        System.exit(0);
```

Comment:
`System.exit(0);` does not guarantee that all Spring `@PreDestroy` callbacks are executed, and it does not guarantee that all running threads are stopped properly.

I would recommend closing the application context first, then `System.exit(0);`.

Reply:
So I tried both the approaches:
1. Tried to close the applicationContext first and then did exit(0). It looks like it is force stopping the routes
```
@Component
public class ShutdownServer {
    @Autowired
    private ApplicationContext applicationContext;

    public void shutdown() {
        SpringApplication.exit(applicationContext);
        System.exit(0);
    }
}
```
![Screenshot 2024-05-13 at 6 01 46 PM](https://github.com/ozone-his/eip-client/assets/36266597/9e3160a4-8f7d-41a7-bff7-7ba9bb5ef72f)

2. Directly did exit(0). It did not log force stopping
```
@Component
public class ShutdownServer {
    @Autowired
    private ApplicationContext applicationContext;

    public void shutdown() {
        System.exit(0);
    }
}
```
![Screenshot 2024-05-13 at 6 07 54 PM](https://github.com/ozone-his/eip-client/assets/36266597/8d7d5d2d-4bd0-446c-a06f-2549d4605138)

Reply:
It's generally preferable and good practice to let Spring handle the shutdown process of an application rather than abruptly terminating it using System.exit().